index.bs

```diff
@@ -5237,30 +5237,30 @@ partial interface MLGraphBuilder {
     : <dfn>scales</dfn>
     ::
         A list of length 2.
-        Specifies the scaling factor in each spatial dimensions of the input: *[scaleHeight, scaleWidth]*.
+        Specifies the scaling factor in each dimensions of the input in {{MLResample2dOptions/axes}} : *[scaleForFirstAxis, scaleForSecondAxis]*.
         The default value is [1.0, 1.0].
 
     : <dfn>sizes</dfn>
     ::
         A list of length 2.
-        Specifies the target sizes for each spatial dimensions of the input: *[sizeHeight, sizeWidth]*. When the target sizes are specified, the {{MLResample2dOptions/scales}} argument is ignored, since the scaling factor values are derived from the target sizes of each spatial dimension of the input.
+        Specifies the target sizes for each dimensions of the input in {{MLResample2dOptions/axes}} : *[sizeForFirstAxis, ForSecondAxis]*. When the target sizes are specified, the {{MLResample2dOptions/scales}} argument is ignored, since the scaling factor values are derived from the target sizes of the input.
 
     : <dfn>axes</dfn>
     ::
         A list of length 2.
-        Specifies the two consecutive dimensions of the input tensor to which the interpolation algorithm applies. The valid values are [0, 1], [1, 2] or [2, 3].
+        Specifies the two dimensions of the input tensor to which the interpolation algorithm applies.
         The default value is [2, 3].
 </dl>
 
 <details open algorithm>
   <summary>
-    To <dfn for="MLGraphBuilder">check resample options</dfn> given |options|, run the following steps:
+    To <dfn for="MLGraphBuilder">check resample options</dfn> given |options| and |input|, run the following steps:
   </summary>
     1. If |options|.{{MLResample2dOptions/scales}} does not [=map/exist=], set it to the [=/list=] « 1.0, 1.0 ».
     1. Otherwise, if any of its values is not greater than 0, or if its [=list/size=] is not 2, return false.
     1. If |options|.{{MLResample2dOptions/sizes}} [=map/exists=], and if its size is not 2, or if any of its values is not greater than 0, return false.
     1. If |options|.{{MLResample2dOptions/axes}} does not [=map/exists=], set it to the [=/list=] « 2, 3 ».
-    1. Otherwise, if its value is not one of « 0, 1», « 1, 2», « 2, 3 », return false.
+    1. Otherwise, if |options|.{{MLResample2dOptions/axes}} contains duplicate values, or if any of its elements is not in [=the range=] 0 to |input|'s [=MLOperand/rank=], exclusive, then return false.
```

Comment:
The elegant aspect I like about this simple check (rather than a hard-coded list of permutations) is that it's more consistent with most other operators, and it also easily permits (in the future) us to relax resample2d to accept pure 2D or 3D rank input, so the caller doesn't need to add extra reshape calls just to prepend leading 1's for the sake of 4Dness. Granted, 2 of 3 current backends require 4Dness anyway ([TF](https://www.tensorflow.org/api_docs/python/tf/image/resize), [CoreML](https://apple.github.io/coremltools/source/coremltools.converters.mil.mil.ops.defs.html#coremltools.converters.mil.mil.ops.defs.iOS15.image_resizing.resample), [DML](https://learn.microsoft.com/en-us/windows/win32/api/directml/ns-directml-dml_resample_operator_desc))